server/pull_request.go

```diff
@@ -426,26 +426,26 @@ func (s *Server) CleanOutdatedPRs() {
 		prs = result.Data.([]*model.PullRequest)
 	}
 
-	mlog.Info("Will process the PRs", mlog.Int("PRs Count", len(prs)))
+	mlog.Info("Processing PRs", mlog.Int("PRs Count", len(prs)))
 
 	for _, pr := range prs {
-		pull, _, errPull := s.GithubClient.PullRequests.Get(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number)
-		if errPull != nil {
-			mlog.Error("Error getting Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number), mlog.Err(errPull))
-			if _, ok := errPull.(*github.RateLimitError); ok {
-				mlog.Error("GitHub rate limit reached")
-				s.CheckLimitRateAndSleep()
-			}
+		pull, _, err := s.GithubClient.PullRequests.Get(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number)
```

Comment:
Instead of getting all individual PRs, the code should bulk fetch them and calculate the diff locally. The is of cause more logic but it greatly reduces the number of request. The same comment applies to `CheckPRActivity`.

Reply:
I want to address the underlying issue next. But unfortunately this has to do for now.

Reply:
Sure, but the code changes doesn't change the number of API call that mattermod does. It only smooths them over a longer time period. Given that the rate limit failure happen all day I don't think this approach helps much.